File: apps/worker/src/aurum_worker/models/safety.py
"""Fixed Bootstrap safety policy."""

from typing import Literal

from aurum_worker.models.base import MaximumPermittedVolume, WireModel


class BootstrapSafetyPolicy(WireModel):
    """Compile-time and runtime locked Demo policy."""

    schema_version: Literal[1]
    environment: Literal["DEMO_ONLY"]
    runtime_mode: Literal["shadow"]
    canonical_symbol: Literal["XAUUSD"]
    maximum_permitted_volume: MaximumPermittedVolume
    maximum_open_positions: Literal[1]
    stop_loss_required: Literal[True]


BOOTSTRAP_SAFETY_POLICY = BootstrapSafetyPolicy.model_validate(
    {
        "schemaVersion": 1,
        "environment": "DEMO_ONLY",
        "runtimeMode": "shadow",
        "canonicalSymbol": "XAUUSD",
        "maximumPermittedVolume": 0.01,
        "maximumOpenPositions": 1,
        "stopLossRequired": True,
    }
)

File: packages/contracts/src/safety.ts
import { z } from "zod";

export const BootstrapSafetyPolicySchema = z
  .object({
    schemaVersion: z.literal(1),
    environment: z.literal("DEMO_ONLY"),
    runtimeMode: z.literal("shadow"),
    canonicalSymbol: z.literal("XAUUSD"),
    maximumPermittedVolume: z.literal(0.01),
    maximumOpenPositions: z.literal(1),
    stopLossRequired: z.literal(true),
  })
  .strict();

export type BootstrapSafetyPolicy = z.infer<typeof BootstrapSafetyPolicySchema>;

export const BOOTSTRAP_SAFETY_POLICY = Object.freeze({
  schemaVersion: 1,
  environment: "DEMO_ONLY",
  runtimeMode: "shadow",
  canonicalSymbol: "XAUUSD",
  maximumPermittedVolume: 0.01,
  maximumOpenPositions: 1,
  stopLossRequired: true,
} satisfies BootstrapSafetyPolicy);

BootstrapSafetyPolicySchema.parse(BOOTSTRAP_SAFETY_POLICY);
